ext/pcre/php_pcre.stub.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/** @return int|false */
+function preg_match(string $pattern, string $subject, array &$subpatterns, int $flags = 0, int $offset = 0) {}
```

Comment:
Here and elsewhere: In PHP the type on reference arguments is the type that the reference needs to have *on entry*, not what it will have on exit. All of these functions don't have a requirement for the entry value, so `&$subpatterns` should have no type.

Additionally, `$subpatterns` is an optional argument, so it should have a default value. As there is no "real" default value in this case, we use `&$subpatterns = UNKNOWN` by convention (though I guess just `null` would work here as well).

Reply:
Ah, sorry I missed adding the default null on `$subpatterns`

For the reference params, what would you like set as the default for the `&$count`  params? They get initialised as `0` in the `.c`, but I can set them as null too, for consistency if thats the standard?

Reply:
I guess `null` is a reasonable default for reference parameters. That's what you'd probably put there in user code.